AGENTS.md

```diff
@@ -0,0 +1,495 @@
+agent = Agent(model=bedrock_model, tools=[...])
+```
+
+### 4. MCP Integration
+
+```python
+from strands import Agent
+from strands.tools.mcp import MCPClient
+from mcp import stdio_client, StdioServerParameters
+
+# Connect to MCP server
+mcp_client = MCPClient(
+    lambda: stdio_client(
+        StdioServerParameters(
+            command="uvx", 
+            args=["awslabs.aws-documentation-mcp-server@latest"]
+        )
+    )
+)
+
+with mcp_client:
+    agent = Agent(tools=mcp_client.list_tools_sync())
+    result = agent("Tell me about Amazon Bedrock")
+```
+
+### 5. Multi-Agent Systems
+
+```python
+from strands.multiagent import Swarm, GraphBuilder
+
+# Swarm pattern
+swarm = Swarm([researcher_agent, calculator_agent, writer_agent])
+result = swarm.run("Research AI market size and calculate growth projections")
+
+# Graph pattern
+graph = (GraphBuilder()
+    .add_agent("researcher", researcher_agent)
+    .add_agent("calculator", calculator_agent)
+    .add_edge("researcher", "calculator")
+    .build())
+```
+
+### 6. Session Management
+
+```python
+from strands.session import FileSessionManager, S3SessionManager
+
+# File-based persistence
+file_session = FileSessionManager(base_path="./sessions")
+
+# S3-based persistence
+s3_session = S3SessionManager(
+    bucket_name="my-agent-sessions",
+    region="us-west-2"
+)
+
+agent = Agent(session_manager=file_session)
+```
+
+---
+
+## Advanced Features
+
+### Hot Reloading Development
+
+```python
+# Enable automatic tool reloading from ./tools/ directory
+agent = Agent(load_tools_from_directory=True)
+
+# Tools in ./tools/ directory are automatically loaded and reloaded
+# when files change during development
+```
+
+### Structured Output
+
+```python
+from pydantic import BaseModel
+
+class AnalysisResult(BaseModel):
+    sentiment: str
+    confidence: float
+    key_topics: list[str]
+
+# Get type-safe structured output
+async for event in agent.model.structured_output(
+    output_model=AnalysisResult,
+    prompt=[{"role": "user", "content": "Analyze this text..."}]
+):
+    if isinstance(event.get("output"), AnalysisResult):
+        result = event["output"]
+        print(f"Sentiment: {result.sentiment}")
+```
+
+### Event Hooks System
+
+```python
+from strands.hooks import HookProvider, HookRegistry
+from strands.hooks.events import BeforeInvocationEvent, AfterInvocationEvent
+
+class LoggingHooks(HookProvider):
+    def register_hooks(self, registry: HookRegistry) -> None:
+        registry.add_callback(BeforeInvocationEvent, self.log_start)
+        registry.add_callback(AfterInvocationEvent, self.log_end)
+
+    def log_start(self, event: BeforeInvocationEvent) -> None:
+        print(f"Request started for agent")
+
+    def log_end(self, event: AfterInvocationEvent) -> None:
+        print(f"Request completed for agent")
+
+agent = Agent(hooks=[LoggingHooks()])
+```
+
+### Concurrent Tool Execution
+
+```python
+from strands.tools.executors import ConcurrentToolExecutor
```

Comment:
[This is the default](https://strandsagents.com/latest/documentation/docs/user-guide/concepts/tools/executors/), so this isn't really "advanced features"; we could show `SequentialToolExecutor` instead